app/services/search/base_service.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Search
+  class BaseService
```

Comment:
This implantation does not follow idiomatic Ruby 😅 This more like Java or Kotlin OOP style.

A more Ruby-ish way is to use the factory pattern:

```suggestion
  class SearchService
    def initialize(search_engine:)
      case 'google'
        Search::Google.new(keyword, language
      case 'bing'
        Search::Bing.new(keyword, language
      else
        raise NotImplementedError
      end
    end
  end
```

Reply:
By refactoring to apply the factory pattern, I moved the search function to `GoogleSearchService` class, the same will be added for BingSearchService.